### PR TITLE
Add package.xml

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary

Add `package.xml` based on dependencies found in `.github/ci/packages`. 

This is to test out using vendor packages to provide Gazebo packages to ROS users (see https://github.com/gazebo-tooling/release-tools/issues/1117). 

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by azeey** [comment]
@scpeters This is ready for review.

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="2">
+  <name>gz-math7</name>
+  <version>7.4.0</version>
+  <description>Gazebo Math : Math classes and functions for robot applications</description>
+  <maintainer email="natekoenig@gmail.com">Nate Koenig</maintainer>
```
Comment (scpeters) [inline package.xml:7]: @adityapande-1995 and I are listed as the [CODEOWNERS](https://github.com/gazebosim/gz-math/blob/gz-math7/.github/CODEOWNERS)

Comment (azeey) [reply package.xml:7]: I copied this from the README. Should we update that as well?

Comment (scpeters) [reply package.xml:7]: yeah, let's update the README as well

Comment (scpeters) [reply package.xml:7]: or remove it from the README

Comment (scpeters) [reply package.xml:7]: I guess we list maintainers in the README of other packages

Comment (scpeters) [reply package.xml:7]: so yes, let's update the readme as well

Comment (azeey) [reply package.xml]: Updated in https://github.com/gazebosim/gz-math/pull/581/commits/3054479a4a60d35c02d0c617950df6076a49af24

**Review by scpeters — APPROVED** [review]

**Pull request merged**